summary_be/archive/utils/summarization.py

```diff
@@ -1,248 +1,50 @@
-
-    Parameters
-    ----------
-    word : string
-    - word to compute score
-    setences : []
-    - array of sentences in text
-
-    Returns
-    -------
-    score : float
-    - idf score of word
-"""
-
-
-def idf_score(word, sentences):
-    num_sent_containing_word = 0
-    for sent in sentences:
-        words = preprocess(sent)
-        if word in words:
-            num_sent_containing_word += 1
-    return math.log10(len(sentences) / num_sent_containing_word)
-
-
-"""
-    Function to return sentence weights based on tf-idf score
-
-    Parameters
-    ----------
-    sentences : []
-    - array of sentences in text
-
-    Returns
-    -------
-    sentence_weight : {}
-    - dictionary of sentence weights
-    - key is index of sentence in sentences
-    - val is tf-idf score
-"""
-
-
-def get_sentence_weights(sentences):
-    sentence_weight = {}
-    for x in range(len(sentences)):
-        sentence = sentences[x]
-        importance = tf_idf_score(sentence, sentences)
-        sentence_weight[x] = importance
-    sentence_weight = dict(
-        sorted(sentence_weight.items(), key=lambda item: item[1], reverse=True)
-    )
-    return sentence_weight
-
-
-"""
-    Function to return indices of top scoring sentences
-
-    Parameters
-    ----------
-    sentence_weight: {}
-    - dictionary of sentence weights
-    num_sent: int
-    - number of top scoring sentences to return
-
-    Returns
-    -------
-    sentence_idx : []
-    - indices of top scoring sentences
-"""
-
-
-def get_top_scoring_sent(sentence_weight, num_sent):
-    curr_num_sentences = 0
-    sentence_idx = []
-    for key in sentence_weight:
-        if key == 0:
-            continue
-        if curr_num_sentences < num_sent - 1:
-            sentence_idx.append(key)
-            curr_num_sentences += 1
-        else:
-            break
-    sentence_idx.sort()
-    return sentence_idx
-
-
-"""
-    Function to summarize a given text.
-
-    Parameters
-    ----------
-    text : string
-    - text to summarize
-    num_sentences : int
-    - number of sentences to include in summary
-
-    Returns
-    -------
-    summary : string
-    - extractive summary of text
-"""
-
-
-def summarize(text, num_sentences):
-    if num_sentences < 1:
-        raise Exception("Summary must have at least one sentence.")
-    sentences = sent_tokenize(text)
-    sentence_weight = get_sentence_weights(sentences)
-    sentence_idx = get_top_scoring_sent(sentence_weight, num_sentences)
-
-    summary = sentences[0]
-    for x in range(len(sentences)):
-        if x in sentence_idx:
-            summary += " "
-            summary += sentences[x]
-
-    return summary
+import pandas as pd
+import numpy as np
+import spacy
+from spacy.language import Language
+import tensorflow as tf
+from tensorflow import keras
+from sentence_transformers import SentenceTransformer
+
+model = tf.keras.models.load_model('/lstm_bi25')
```

Comment:
Should we change the file path to "./lstm_bi25"?

Reply:
yes sorry!

Reply:
the notebooks are in the Essentially google drive 😄 as well as the models. I tried installing tensorflow a few times but it wouldn't work and i gave up. Don't worry, I made sure to include it in the requirements.txt so it should work for you guys!